### ISSUE TITLE
Constraint SR3

### ISSUE THREAD

**Issue by sagastra** [dynamicslab/pysindy#224]
Hi,
I am trying to use the constrained SR3 to solve a system of equations. I have to set the constraints on the diagonal to -1, and I used to run code that did just that 

n_vox = 66
#contraints
rhs = np.array([-1 for i in range(n_vox)])
lhs = np.zeros((n_vox, n_vox * n_features))

for i in range(n_vox):
    lhs[i, i+i*n_features]=1

but the constraints after solving are set to twice that at -2. I am confused on why this is happening, maybe I am not thinking about this correctly. Thank you

**Comment by akaptano** [comment]
Hi sagastra,

Could you clarify what you mean by "the constraints after solving are set to twice that at -2"? Are you saying that the coefficients that you constrained to be -1 are being identified by SINDy as having a value of -2? 

It is not even clear what you are intending to do with the constraints. Do you want to the diagonal of the coefficient matrix to sum to -1 (a constraint on the trace of the coefficient matrix) or set each of the diagonal values of the coefficient matrix to a value of -1? If it is the latter, I believe you should be using "lhs[i, i*n_features]=1" instead of "lhs[i, i+i*n_features]=1".

**Comment by sagastra** [comment]
Hi akaptano,
Thanks so much for the quick response. Ya, when I print the diagonal (which I am trying to set the constraint to -1), it prints out -2 instead. No lhs[i, i*n_features]=1, gives me a completely wrong answer. The constraints are set in place where they should be (on the diagonal each -1), but are double somehow at -2.

**Comment by akaptano** [comment]
Could you print out the first few rows of the lhs matrix? And are the constrained coefficients the only coefficients that are doubled? What about the unconstrained coefficients?

**Comment by sagastra** [comment]
Okay, I tried it on my old laptop and the same code was giving me the correct constraints, even when I updated the sindy to the latest version. Do you think it might be a python version or another package that might interact in a strange way with sindy?

**Comment by Jacob-Stevens-Haas** [comment]
@sagastra, to look for a strange interaction, we would need a complete, minimal working example (MWE).  Without that, it's a lot 

Very often, in trying to pare down a question to an MWE and demonstrate that the MWE behaves exactly as described, people solve their own issue or identify the issue directly.  Package conflicts are possible, but far rarer than user error.  See the [Stackoverflow guidance](https://stackoverflow.com/help/minimal-reproducible-example) on how to build an MWE for your question.

It also helps to enclose your code in these messages with triple backticks like this:
\`\`\`python
foo = "bar"
\# more code
\`\`\`

It renders as:
```python
foo = "bar"
# more code
```

**Comment by sagastra** [comment]
Okay I figured out what was happening. (I am using sindy 1.3. and python 3.7). For constrainedSR3 the variable threshold is multiplied times my constraints. If I change the threshold for the call on the optimizer to 0.5 then the constraints I set show up correctly. If I change it to any other number it multiplies times the constraint I have.  (i.e 0.05 is 1/10 to what I set the constraints are). Basically I think there might be a masking problem, or maybe I am using this incorrectly.

I  

optimizer = ps.ConstrainedSR3(constraint_rhs=rhs, constraint_lhs=lhs,  threshold =0.5, nu=100,  thresholder= "l2")
